daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ledgerinteraction/GrpcLedgerClient.scala

```diff
@@ -322,7 +320,9 @@ class GrpcLedgerClient(val grpcClient: LedgerClient, val applicationId: Applicat
       val events = transactionTree.getTransaction.rootEventIds
         .map(evId => transactionTree.getTransaction.eventsById(evId))
         .toList
-      events.traverse(fromTreeEvent(_)) match {
+      events.zip(commandResultPackageIds).traverse { case (event, intendedPackageId) =>
```

Comment:
I'm not very happy about this `zip` and the one in `fromTransactionTree`, especially considering 
> Note that CreateAndExerciseCommand gives two results, so we duplicate the package id

but I don't have a better idea atm :(

Reply:
yeah it is somewhat upsetting, if each command had some kind of ID that the results referred back to, we could create a small lookup

Reply:
I have no idea if this exists though, I assume it doesn't